torch/_jit_internal.py

```diff
@@ -647,7 +646,12 @@ class _IgnoreContextManager(contextlib.AbstractContextManager):
     def __init__(self, **kwargs):
         pass
 
-    def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any) -> None:
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
```

Comment:
Seems like torch jit doesn't support BaseException, but it may support Exception? I worry it doesn't support traceback type either though.

Reply:
yeah it might but if we switch to `Exception` the PYI036 check will fail again with `PYI036 The first argument in `__exit__` should be annotated with `object` or `type[BaseException] | None`` error. Only `BaseException` will pass the check.

Reply:
I guess we could do the same as [PYI041](https://github.com/pytorch/pytorch/pull/111112) to keep PYI036 disabled? There's no point to #noqa all of them